### DISCUSSION TITLE
audio source switching

### DISCUSSION THREAD

**Discussion by alicealysia** [Elvyria/Mixxc#5] [Ideas]
Hi! Just wanted to make a quick feature request. I'd like the ability to switch audio sources. Unless that feels a little out of scope of course.

Other than that, this is fantastic! I love that it uses the shell layer, and it's the perfect addition to my waybar and is really helping my setup feel less like a mishmash of random apps, and more like a proper desktop environment.

**Comment by Elvyria** [comment]
Any ideas how would you like this implemented UX wise ?

My current flow is just switching between sources with a keyboard shortcut, so I personally don't have a need for it, but it should be trivial to do, aside from figuring out how and where to shove that.

**Reply by alicealysia** [reply]
You could possibly add an icon to the right of the master volume slider, which when clicked could either open a drop down menu, or could cause the volume mixer to slide to the left and hide itself, to reveal a scrollable list of sources (this would involve significant amounts of overflow: hidden abuse, this is how windows 11 does it btw... but with ugly volume slider icons lol)

**Reply by Elvyria** [reply]
![image](https://github.com/user-attachments/assets/33a85296-bf5b-46f2-8de7-a64a1cc3f49c)
I have an idea for a slim bar to the side, which would have hardware sources as icons with some assumptions by source names (HDMI,  etc), assuming that no one is insane enough to use it to switch between more than like 3 sources.

Do you think that would be more suitable ? If you can, you can just draw what you have in mind with boxes for ideas, it's okay.

**Reply by alicealysia** [reply]
Yeah, that's a great idea!

**Reply by Elvyria** [reply]
Okay, I'll add it to the project board. I'll probably make a 0.2.3 soon to push sound popping fix on launch and you can expect source switch in 0.2.4, if I don't find it too problematic to implement for some reason.

**Comment by Elvyria** [comment]
I have pushed the first iteration of a feature in 650b272, you can build it and try it out, if you're unable to build it yourself, I can provide you with the binary.

Don't forget to update your styles with the new [elements](https://github.com/Elvyria/Mixxc/blame/650b2725009a3ff26759d5f22ef28f51a5e55ece/style/default.css#L19) or remove existing one and the default will be recreated.

Please report any issues encountered in #6.

**Reply by Elvyria** [reply]
I'm aware of the issue when output visually doesn't change if nothing is playing, it will be fixed.

**Reply by alicealysia** [reply]
thank you! I've built and installed the latest commit, so I'll keep an eye  out for the next, as well as any bugs outside the known one.

**Reply by Elvyria** [reply]
c28c1d0 should be now suitable for a proper feature testing.

**Comment by Elvyria** [comment]
Implemented in [0.2.4](https://github.com/Elvyria/Mixxc/releases/tag/0.2.4), can be hidden and stylized via CSS.
